### PR TITLE
13 google oauth

### LINKED ISSUE #13: Google OAuth
Setup Google OAuth using Omni Auth to allow users to login

### PR DESCRIPTION
**Pull request by exasperlnc**
## Changes
- This PR closes #13 Google OAuth
- Adds current user method to Application Controller
- Sets up google OAuth using OmniAuth

## Type of Changes
- [ ] new feature
- [x] setup
- [x] chore
- [ ] bug fix
- [ ] refactor
- [ ] other: 

## Checklist
- [ ] Tests added for new functionality (No easy way to test omniAuth apparently, so just tested by hand) 
- [x] All other tests are also passing (please note if not) 
  - Current coverage %: n/a

## Number of reviewers
- [x] 1
- [ ] 2
- [ ] 3

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get 'auth/google_oauth2/callback', to: 'sessions#create'
+  get '/dashboard', to: 'dashboard#show'
```
Comment (cariperi) [inline config/routes.rb:7]: No need to address in this PR, but let's revisit front end route names as a group to decide exactly what we want them to be from the user's perspective.

Comment (cariperi) [reply config/routes.rb:7]: @exasperlnc @SMcPhee19

**Comment by cariperi** [comment]
Thanks for getting this up and running! It's good to merge without changes, just leaving up so you can see the comment.

File: config/routes.rb (hunk 1 above)
Comment (exasperlnc) [reply config/routes.rb:7]: Sounds like a plan! We can do it during tomorrow's Standup? I'll add it to the miro board!

**Review by cariperi — APPROVED** [review]

**Pull request merged**